ts/model_service_worker.py

```diff
@@ -53,9 +62,13 @@ def __init__(self, s_type=None, s_name=None, host_addr=None, port_num=None):
         logging.info("Listening on port: %s", s_name)
         socket_family = socket.AF_INET if s_type == "tcp" else socket.AF_UNIX
         self.sock = socket.socket(socket_family, socket.SOCK_STREAM)
+        self.metrics_cache = MetricsCacheYamlImpl(config_file_path=metrics_config)
+        if self.metrics_cache:
+            self.metrics_cache.initialize_cache()
```

Comment:
Is a separate initialize_cache necessary or can we call this from constructor? In which case would self.metrics_cache be False/None and RuntimeError would be raised here?

Reply:
Agreed. I left it the way it was in the initial implementation. I don't mind changing it, although it would require a bunch of changes to the unit tests since there are test cases for adding custom metrics before the cache is initialized. cc @lxning

Reply:
@maaquib let's create a ticket for this minor change in next release.

Reply:
https://github.com/pytorch/serve/issues/1966